### ISSUE TITLE
(skflow) issue in batch normalization after restore

### ISSUE THREAD

**Issue by laouer** [tensorflow/tensorflow#2363]
Hi,
I'm using Tensorflow [last succeed build #85](http://ci.tensorflow.org/view/Nightly/job/nigntly-matrix-linux-gpu/TF_BUILD_CONTAINER_TYPE=GPU,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=gpu-working/lastSuccessfulBuild/)

I succeed to train a classifier with batch normalization and saving it with classifier.save(logdir) and restoring the classifier with 
classifier = tf.contrib.skflow.TensorFlowEstimator.restore(logdir)

the classifier is restored but when using prediction classifier.predict(X_test) it seems that the batch normalization is applied as is in Training mode.

Looking to the code of batch normalization (batch_norm_ops.py)
`is_training = array_ops_.squeeze(ops.get_collection("IS_TRAINING"))
    mean, variance = control_flow_ops.cond(is_training, update_mean_var,
                                           lambda: (ema_mean, ema_var))`

I suppose as the graph is loaded in restore step, the "IS_TRAINING" is defined at the state of True at the saved step. and the cond function used will be update_mean_var at predict step , however the "IS_TRAINING" is changed at _predict step.
you can detect the problem when classifying 2 class problem and adding a batch normalization before the softmax the mean of the softmax of any tested data will be 0.

Maybe a solution will be to use a placeholder in place of the variable is_training

Or I'm missing something ?

**Comment by imcomking** [comment]
Hello.
Actually this is an equivalent problem with next issue.

https://github.com/tensorflow/tensorflow/issues/2167

These problems are caused by unsaved graph collections.

**Comment by girving** [comment]
@ilblackdragon: Is this a duplicate of #2167?  If so please close as such.

**Comment by ilblackdragon** [comment]
Should be fixed now with current way of restoring via recreating the graph.
